openquake/hazardlib/geo/surface/simple_fault.py

```diff
@@ -189,12 +189,10 @@ def get_fault_patch_vertices(cls, fault_trace, upper_seismogenic_depth,
         """
         # Similar to :meth:`from_fault_data`, we just don't resample edges
         dip_tan = math.tan(math.radians(dip))
-        hdist_top = upper_seismogenic_depth / dip_tan
-        hdist_bottom = lower_seismogenic_depth / dip_tan
-
+        hdist_bottom = (
+            lower_seismogenic_depth - upper_seismogenic_depth) / dip_tan
```

Comment:
To obtain the bottom_edge_point (at line 206), we need the parameter hdist_bottom.
To obtain the horizontal distance between the top_edge_point  and bottom_edge_point, we should calculate by the depth change (lower_seismogenic_depth - upper_seismogenic_depth) between the two edge points, instead of taking directly the lower_seismogenic_depth value, to get the correct rupture vertices. 

Before this debugging, the location of  rupture plane calculated from the method was shifting deeper than the actual rupture plane we set up. It will not change the final result on the pure strike slip(with dip = 90 degree), since the plane equation is the same. However, the error will show up when we calculate the dipping fault case.